### ISSUE TITLE
Reduce Lint Warnings from Generated JavaScript

### ISSUE THREAD

**Issue by jednano** [microsoft/TypeScript#765]
I'm using [CodeClimate](https://codeclimate.com/) to monitor the quality, style and test coverage of my source files. I wish [CodeClimate](https://codeclimate.com/) could monitor TypeScript files directly, but until that day comes, I have it monitoring my [TypeScript-generated JavaScript source files](https://github.com/blinkjs/blink/tree/master/js) for my project, [blink](https://github.com/blinkjs/blink). This is, of course, why I'm getting a number of lint warnings flagged as code smells, some of which could be easily resolved. For example, this warning:

> A constructor name should start with an uppercase letter.
> 
> ``` js
> d.prototype = new __();
> ```

If the `extends` code was change from:

``` js
var __extends = this.__extends || function(d, b) {
```

to:

``` js
var __extends = this.__extends || function(D, b) {
```

That particular warning would go away. Of course, this is just one example of perhaps only a handful of lint warnings, but I feel that TypeScript-generated JavaScript should do its best to ensure these lint warnings aren't just noise on [a CodeClimate report](https://codeclimate.com/github/blinkjs/blink/code?sort=smell_count&sort_direction=desc) or any other report of this nature.

Another lint warning I see often is:

> Wrapping non-IFFE function literals in parens is unnecessary.
> 
> ``` js
> var override = (function (config) {
> ```

**Comment by valorkin** [comment]
@jedmao how did you did it?
I just tweeted, need to show how much TS is used now:
https://twitter.com/valorkin/status/686197003743698945
And how we will appreciate support of it :)
